### PR TITLE
Add shareHash parameter

### LINKED ISSUE #1468: API documentation is missing `shareHash`
**Describe the bug**
The API documentation for inserting a submission is missing the required `shareHash` parameter.
Without the `shareHash` parameter users can not insert submission on public shares (if they are not the owner of the form).

### PR DESCRIPTION
**Pull request by Chartman123**
Fixes #1468 

Add missing shareHash parameter to documentation of  `/api/v2/submission/insert`

Signed-off-by: Chartman123 <chris-hartmann@gmx.de>

### CODE REVIEW

File: docs/API.md [hunk 1]
```diff
@@ -540,6 +541,8 @@ Store Submission to Database
   |-----------|---------|-------------|
   | _formId_  | Integer | ID of the form to submit into |
   | _answers_ | Array   | Array of Answers |
+  | _shareHash_ | String | Hash of a public link share, else `''` |
```
[the remark this answers is not in the document]
Comment (Chartman123) [reply docs/API.md]: Better now?

[the remark this answers is not in the document]
Comment (jotoeri) [reply docs/API.md]: Hmm, a bit, yes. Still i have a bit in mind the confusion, that a 'publicly shared form' can be publicly shared, but you don't submit through the public link. So from the text you would need to provide a hash once a form just _has_ a public link. But indeed, thats i-tüpfeles scheißerei. 😜

**Review by jotoeri — APPROVED** [review]
I leave it up to you. 😉 👍

**Review by susnux — APPROVED** [review]
better :smiley:

**Pull request merged**